src/ngraph_encapsulate_clusters.cc

```diff
@@ -482,7 +486,24 @@ Status EncapsulateClusters(Graph* graph, int graph_id) {
     graph->RemoveNode(node);
   }
 
-  // Pass 7 (optional, only run if environment variable
+  // Pass 7: Insert to function library
+  // Note: We loop over cluster_indices_for_this_rewrite and not all the
+  // contents of ClusterManager
+  for (const auto& cluster_idx : cluster_indices_for_this_rewrite) {
+    // TODO: whats the right flib to use in sgraph's constructor?
+    Graph sgraph(graph->flib_def());
+    // TODO: When this works, NGraphClusterManager can go away
+    TF_RETURN_IF_ERROR(ConvertGraphDefToGraph(
+        GraphConstructorOptions(),
+        *(NGraphClusterManager::GetClusterGraph(cluster_idx)), &sgraph));
+    FunctionDef* fdef = fdeflib->add_function();
+    // TODO: if func lib has func with same name etc?
+    TF_RETURN_IF_ERROR(GraphToFunctionDef(
+        sgraph, strings::StrCat("ngraph_cluster_", to_string(cluster_idx)),
+        fdef));
```

Comment:
TODO: add comment about the magic happening here.
Basically its:
graphdef(NGraphClusterManager::GetClusterGraph(cluster_idx)) --> graph(sgraph) --> functiondef(fdef)